### DISCUSSION TITLE
Found issue in Error Logging

### DISCUSSION THREAD

**Discussion by jayeshtejwani** [jamessimone/apex-rollup#299] [Q&A] [answered]
Hi @jamessimone,

Starting as a discussion but may be this is an issue. 
I am running the rollups on parent object Account. I have written a custom logging class to log catch the errors thrown during Account Update in our custom object. To test if errors are being logged, I create null pointer exception in AccountTriggerHandler(before update), and that's seen in debug logs but not caught or displayed by RollUp Logger or Custom Logger class, which doesn't help.

**Extended Class**
![Screen Shot 2022-04-20 at 12 21 39 AM](https://user-images.githubusercontent.com/18471403/164150070-b6b4a726-d0cc-402f-9943-0640ff61efa9.png)

**Debug Logs from the Custom Logger Debug Statement**
![Screen Shot 2022-04-20 at 12 22 35 AM](https://user-images.githubusercontent.com/18471403/164150128-ecbe30d9-48f8-4b9a-ad9b-2e76ae647865.png)
![Screen Shot 2022-04-20 at 12 22 48 AM](https://user-images.githubusercontent.com/18471403/164150131-35df3d84-00ed-4486-b0d3-8889249f28b2.png)
![Screen Shot 2022-04-20 at 12 23 01 AM](https://user-images.githubusercontent.com/18471403/164150134-15e39d0b-de85-4e0f-8292-54cea7abe16e.png)

**Success Printed for a failed transaction. Debug Logs of the FlowOutput(received after List<Rollup.FlowOutput> innerOutputs = Rollup.performRollup(flowInputList);)**

![Screen Shot 2022-04-20 at 12 28 57 AM](https://user-images.githubusercontent.com/18471403/164150731-157ee501-514e-4084-aea7-b87c0886dc61.png)

**Null Pointer exception**

![Screen Shot 2022-04-20 at 12 23 29 AM](https://user-images.githubusercontent.com/18471403/164150314-ba3a2d65-36da-4b21-861b-8f0878391975.png)

**Comment by jamessimone** [comment, marked as answer]
@jayeshtejwani the default behavior of the included logger is to only print messages to the apex debug logs. I would look at [the implementation for the actual Custom Object logger](https://github.com/jamessimone/apex-rollup/blob/main/plugins/CustomObjectRollupLogger/classes/RollupCustomObjectLogger.cls) if you’re interested in seeing how to save the logged inputs to a custom object. Not all logging within Apex Rollup is going to be logged in the event of an uncaught exception, but the example you’re using is with an exception _outside_ of Apex Rollup, so … it has to be caught/handled outside of Apex Rollup. The library isn’t mean to be an end-all logging solution for the rest of your org, or for problems that occur elsewhere in your flows/Apex. For that, I would highly recommend the usage of [Nebula Logger](https://github.com/jongpie/NebulaLogger).
